src/java.base/share/classes/java/nio/Heap-X-Buffer.java.template

```diff
@@ -278,6 +278,63 @@ class Heap$Type$Buffer$RW$
 
 #if[char]
 
+    //
+    // Use getChars() to load chars directly into the heap buffer array.
+    // For a String or StringBuffer source this improves performance if
+    // a proper subsequence is being appended as copying to a new intermediate
+    // String object is avoided. For a StringBuilder where either a subsequence
+    // or the full sequence of chars is being appended, copying the chars to
+    // an intermedite String in StringBuilder::toString is avoided.
+    //
+    private $Type$Buffer appendChars(CharSequence csq, int start, int end) {
+        checkSession();
+
+        int length = end - start;
+        int pos = position();
+        int lim = limit();
+        int rem = (pos <= lim) ? lim - pos : 0;
+        if (length > rem)
+            throw new BufferOverflowException();
+
+        if (csq instanceof String str) {
+            str.getChars(start, end, hb, ix(pos));
+        } else if (csq instanceof StringBuilder bld) {
```

Comment:
I supposed we rename rename bld  and buf to "sb" so that the naming is consistent with CharBuffer cb, heap buffer hb, etc. No big deal of course but it was sb and seems to have got renamed.

Reply:
There was a little recidivism there; fixed in 72385b8c30646ad2fb2e4399c77d84f0f813f5c3.